### PR TITLE
worked on AddNewCustomer page on designing part and Post and get Api

### PR DESCRIPTION
**Pull request by HarshGenboot123**

### CODE REVIEW

File: src/Type/Components/MapTypes.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    zoomLevel?: number
+}
+
+export interface DisplayPositionProps {
+    map: any
+    onPositionChange: (lat: number, lng: number) => void
+}
+
+export interface DisplayMapProps {
+    center: [number, number]
```
Comment (ShKushal) [inline src/Type/Components/MapTypes.ts:18]: what is this a map ??

File: src/Components/Moormanage/Boatyards/BoatyardDetailsTable.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  return (
+    <div style={{ display: "flex", alignItems: "center" }}>
+      <img
+        src="/assets/images/map.png"
+        alt="sea image"
+        style={{ width: "50vw", height: "200px", marginRight:"20px" , marginTop:"-120px" }}
+      />
+      <div className="float-right bg-[#D9D9D9] rounded-md border-[1px] p-1 border-gray-300 w-[22vw]">
+        <DataTable
```
Comment (ShKushal) [inline src/Components/Moormanage/Boatyards/BoatyardDetailsTable.tsx:22]: change this with custom data table component

File: src/Components/AdminTools/AddNewCustomer.tsx [hunk 3]
```diff
@@ -34,18 +41,23 @@ const AddNewCustomer: React.FC<CustomerAdminDataProps> = ({ customerData, editMo
     }
   }, [editMode, customerData])
 
-  const handleSave = () => {
+  const handleSave = async () => {
     const payload = {
       Name: name,
```
Comment (ShKushal) [inline src/Components/AdminTools/AddNewCustomer.tsx:46]: these payload keys have been updated on backend , please check with pranav on this and make changes are required

**Review by ShKushal — CHANGES_REQUESTED** [review]

File: src/Type/Components/MapTypes.ts (hunk 1 above)
Comment (Sunaina04) [reply src/Type/Components/MapTypes.ts:18]: I created this to develop a seperate component for displaying maps, reducing redundant code.

File: src/Components/Moormanage/Boatyards/BoatyardDetailsTable.tsx (hunk 2 above)
Comment (HarshGenboot123) [reply src/Components/Moormanage/Boatyards/BoatyardDetailsTable.tsx:22]: Not in use, so we are removing this component

File: src/Components/AdminTools/CustomerAdmin.tsx [hunk 4]
```diff
@@ -52,7 +56,19 @@ const Permission = () => {
   const handleModalClose = () => {
     setModalVisible(false)
   }
-  
+
+  const getCustomerData = async () => {
+    try {
+      const response = await getCustomer({}).unwrap()
```
[the remark this answers is not in the document]
Comment (HarshGenboot123) [reply src/Components/AdminTools/CustomerAdmin.tsx:158]: If we remove this curly braces then we get an error Expected 1 arguments, but got 0.ts(2554)
buildHooks.d.ts(355, 6): An argument for 'arg' was not provided.

File: src/Components/AdminTools/AddNewCustomer.tsx (hunk 3 above)
Comment (HarshGenboot123) [reply src/Components/AdminTools/AddNewCustomer.tsx:46]: Done, I have resolved it

File: src/Components/Moormanage/Boatyards/BoatyardDetailsTable.tsx (hunk 2 above)
Comment (ShKushal) [reply src/Components/Moormanage/Boatyards/BoatyardDetailsTable.tsx:22]: why it was present originally ?

Comment (Sunaina04) [reply src/Components/Moormanage/Boatyards/BoatyardDetailsTable.tsx:22]: actaully it was creaed as per figma design , figma was changing again and again

File: src/Type/Components/MapTypes.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+export interface MarkerData {
+    position: [number, number]
```
Comment (ShKushal) [inline src/Type/Components/MapTypes.ts:2]: this

File: src/Type/Components/MapTypes.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    popupContent: string
+}
+
+export interface CustomSelectPositionMapProps {
+    onPositionChange: (lat: number, lng: number) => void
+    center?: [number, number]
```
Comment (ShKushal) [inline src/Type/Components/MapTypes.ts:8]: this

File: src/Type/Components/MapTypes.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    zoomLevel?: number
+}
+
+export interface DisplayPositionProps {
+    map: any
+    onPositionChange: (lat: number, lng: number) => void
```
Comment (ShKushal) [inline src/Type/Components/MapTypes.ts:14]: this

File: src/Type/Components/MapTypes.ts (hunk 1 above)
Comment (ShKushal) [reply src/Type/Components/MapTypes.ts:18]: all of the above can be moved to one type

type MarkerPosition = [number, number]

File: src/Components/Map/DisplayPosition.tsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+import { DisplayPositionProps } from '../../Type/Components/MapTypes'
+
+const DisplayPosition: React.FC<DisplayPositionProps> = ({ map, onPositionChange }) => {
+  const [position, setPosition] = useState(() => map.getCenter())
+
+  const onMove = useCallback(() => {
+    const newPosition = map.getCenter()
+    setPosition(newPosition)
+    if (onPositionChange) {
+      onPositionChange(newPosition.lat, newPosition.lng)
```
Comment (ShKushal) [inline src/Components/Map/DisplayPosition.tsx:14]: add a clean up function here also

**Review by ShKushal — CHANGES_REQUESTED** [review]

File: src/Type/Components/MapTypes.ts (hunk 5 above)
Comment (Sunaina04) [reply src/Type/Components/MapTypes.ts:2]: in map actually they are expecting value like this -> [number , number]

File: src/Type/Components/MapTypes.ts (hunk 6 above)
Comment (Sunaina04) [reply src/Type/Components/MapTypes.ts:8]: same as above

File: src/Type/Components/MapTypes.ts (hunk 7 above)
Comment (Sunaina04) [reply src/Type/Components/MapTypes.ts:14]: this also am doing as per map conditions

Comment (Sunaina04) [reply src/Type/Components/MapTypes.ts:14]: expecting lat , long in props

File: src/Components/Map/DisplayPosition.tsx [hunk 9]
```diff
@@ -16,11 +16,14 @@ const DisplayPosition: React.FC<DisplayPositionProps> = ({ map, onPositionChange
   }, [map, onPositionChange])
 
   useEffect(() => {
-    map.on('move', onMove)
+    const mapMoveListener = () => {
+      handlePositionChange();
```
Comment (ShKushal) [inline src/Components/Map/DisplayPosition.tsx:20]: not correct,
this use effect was oky before

File: src/Components/Map/DisplayPosition.tsx [hunk 10]
```diff
@@ -7,7 +7,7 @@ import { DisplayPositionProps } from '../../Type/Components/MapTypes'
 const DisplayPosition: React.FC<DisplayPositionProps> = ({ map, onPositionChange }) => {
   const [position, setPosition] = useState(() => map.getCenter())
 
-  const onMove = useCallback(() => {
+  const handlePositionChange = useCallback(() => {
     const newPosition = map.getCenter()
     setPosition(newPosition)
     if (onPositionChange) {
```
Comment (ShKushal) [inline src/Components/Map/DisplayPosition.tsx:13]: you just need to return a clean up function here

**Review by ShKushal — CHANGES_REQUESTED** [review]

**Review by ShKushal — APPROVED** [review]

**Pull request merged**